Assets/Plugins/Source/Core/RuntimeConfig.cs

```diff
@@ -0,0 +1,258 @@
+            PlatformFlags = platformFlags;
+            AuthScopeFlags = authScopeFlags;
+            IntegratedPlatformManagementFlags = integratedPlatformManagementFlags;
+            TickBudgetInMilliseconds = tickBudgetInMilliseconds;
+            ThreadAffinity = threadAffinity;
+            IsServer = isServer;
+            AlwaysSendInputToOverlay = alwaysSendInputToOverlay;
+            InitialButtonDelayForOverlay = initialButtonDelayForOverlay;
+            RepeatButtonDelayForOverlay = repeatButtonDelayForOverlay;
+            SendInputDirectlyToSDK = sendInputDirectlyToSdk;
+        }
+
+        /// <summary>
+        /// Uses the values in the given EOSConfig class to create a
+        /// RuntimeConfig struct that contains readonly values with the
+        /// appropriate types.
+        ///
+        /// The result of using this for setting EOS SDK config values is a sort
+        /// of data validation, in that if the corresponding string values are
+        /// not parseable into the expected data type. Future work will be done
+        /// to expose this validation in an appropriately user-friendly manner.
+        ///
+        /// The result of this conversion should be used when assigning the
+        /// values to structures that aid in the creation and initialization of
+        /// the EOS SDK.
+        /// </summary>
+        /// <param name="config">
+        /// The EOSConfig to convert into RuntimeConfig.
+        /// </param>
+        public static implicit operator RuntimeConfig(EOSConfig config)
+        {
+            if (null == config)
+            {
+                throw new ArgumentNullException($"Cannot convert from a null \"EOSConfig\" to \"RuntimeConfig\".");
+            }
+
+            // Use the configure override thread affinity helper function to
+            // set the thread affinity parameter for the RuntimeConfig object.
+            InitializeThreadAffinity threadAffinity = new();
+            config.ConfigureOverrideThreadAffinity(ref threadAffinity);
+
+            // try parse the float values from string so that they default to
+            // zero if the strings are empty.
+            _ = float.TryParse(config.initialButtonDelayForOverlay, out float initialButtonDelayForOverlayFloat);
+            _ = float.TryParse(config.repeatButtonDelayForOverlay, out float repeatButtonDelayForOverlayFloat);
+
+            return new RuntimeConfig(
+                Guid.Parse(config.productID),
+                config.productName,
+                Version.Parse(config.productVersion),
+                Guid.Parse(config.sandboxID),
+                Guid.Parse(config.deploymentID),
+                new ClientCredentials() { ClientId = config.clientID, ClientSecret = config.clientSecret },
+                config.encryptionKey,
+                config.GetPlatformFlags(),
+                config.GetAuthScopeFlags(),
+                IntegratedPlatformManagementFlags.Disabled,
```

Comment:
Is there any way to pull the IntegratedPlatformManagementFlags from the incoming EOSConfig? It probably shouldn't always be disabled.

Reply:
Ultimately, this will be done, however, as we discovered in the course of this work the `EOSConfig` does not actually _have_ those flags as field members. This is a bug in the current implementation. This particular item will be fixed once that bug is resolved.

Reply:
Marking this conversation as "unresolved," because I chose to include the changes to include the `IntegratedPlatformManagementFlags` field member being added to `EOSConfig` for the sake of clarity. I unresolved it for visibility, feel free to re-resolve it if you're satisfied with this solution.